inc/html.class.php

```diff
@@ -6548,4 +6573,85 @@ static function getInvertedColor($hexcolor = "", $bw = true, $sbw = true) {
          + str_pad($g, 2, '0', STR_PAD_LEFT)
          + str_pad($b, 2, '0', STR_PAD_LEFT);
    }
+
+   /**
+    * Compile SCSS styleshet
+    *
+    * @param array $args Arguments. May contain:
+    *                      - v: version to append (will default to GLPI_VERSION)
+    *                      - debug: if present, will not use Crunched formatter
+    *                      - file: filerepresentation  to load
+    *                      - reload: force reload and recache
+    *                      - nocache: do not use nor update cache
+    *
+    * @return string
+    */
+   public static function compileScss($args) {
+      global $GLPI_CACHE;
+
+      $ckey = isset($args['v']) ? $args['v'] : GLPI_SCHEMA_VERSION;
+
+      $scss = new Compiler();
+      if ($_SESSION['glpi_use_mode'] != Session::DEBUG_MODE && !isset($args['debug'])) { // mode debug
+         $scss->setFormatter('Leafo\ScssPhp\Formatter\Crunched');
+      }
+
+      $paths = [
+         GLPI_ROOT . '/css/',
+         GLPI_ROOT . '/css/palettes/',
+      ];
+
+      if (!isset($args['file']) || $args['file'] == 'main_styles') {
+         $import = '@import "styles.scss";';
+         if (isset($_SESSION['glpihighcontrast_css'])
+            && $_SESSION['glpihighcontrast_css']) {
+            $ckey .= '_highcontrast';
+            $import .= '@import "highcontrast";';
+         }
+
+         // CSS theme
+         $theme = 'auror';
+         if (isset($_SESSION["glpipalette"])) {
+            $ckey .= '_' . $_SESSION['glpipalette'];
+            $theme = $_SESSION['glpipalette'];
+         }
+         $import .= '@import "'.$theme.'";';
+      } else {
+         $exploded = explode('/', $args['file']);
+         $file = array_pop($exploded);
+         if (count($exploded) == 1) {
+            $path = realpath(GLPI_ROOT . '/plugins/' . $exploded[0] . '/css');
```

Comment:
Some plugins does not put their css files into the css directory (our example plugin does not https://github.com/pluginsGLPI/example/blob/master/setup.php#L167 ).

Reply:
Well, they should :D

See also https://glpi-plugins.readthedocs.io/en/latest/empty/index.html#minifying-css-ans-js